### DISCUSSION TITLE
What is the JWT implementation in jwt_auth.py used for?

### DISCUSSION THREAD

**Discussion by gpspake** [getredash/redash#6470] [General]
I'm investigating how we can modify Redash to get a jwt from Auth0 to authenticate connections.

We currently have Redash configured to log in with the Auth0 SAML2 add on - per the docs - but because it's SAML, we don't get the token we need to talk to our APIs (The plan is to figure out how to get a JWT token and include it with requests using a custom data source for our internal stuff). We still haven't figured out how to get a JWT based on the SAML2 Auth0 authentication so if anyone has ideas, I'd love to hear them... but the reason I'm posting is because I'm curious about the existing  JWT implementation in jwt_auth.py and what it's used for. There's nothing anywhere in the docs or existing discussions about it and searching for jwt returns no results.

**Comment by HomiGrotas** [comment]
Hi @gpspake, what you ended up with?

**Comment by arikfr** [comment]
At this point I don't fully remember, but I think it was meant for custom authentication methods. It doesn't seem like it's what you're after.

There are other things you need to consider if you go down the path of having each user they own token, like the shared query results cache.
